src/vaults/factories/PoolEscrowFactory.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {Auth} from "src/misc/Auth.sol";
+
+import {IPoolEscrowProvider, IPoolEscrowFactory} from "src/vaults/interfaces/factories/IPoolEscrowFactory.sol";
+import {IPoolEscrow} from "src/vaults/interfaces/IEscrow.sol";
+import {PoolEscrow} from "src/vaults/Escrow.sol";
+
+contract PoolEscrowFactory is IPoolEscrowFactory, Auth {
+    address public immutable root;
+    address public poolManager;
+    address public balanceSheet;
+    address public asyncRequests;
+
+    mapping(uint64 poolId => address) public escrows;
+
+    constructor(address root_, address deployer) Auth(deployer) {
+        root = root_;
+    }
+
+    /// @inheritdoc IPoolEscrowFactory
+    function file(bytes32 what, address data) external auth {
+        if (what == "poolManager") poolManager = data;
+        else if (what == "balanceSheet") balanceSheet = data;
+        else if (what == "asyncRequests") asyncRequests = data;
+        else revert FileUnrecognizedParam();
+        emit File(what, data);
+    }
+
+    /// @inheritdoc IPoolEscrowFactory
+    function newEscrow(uint64 poolId) public auth returns (address) {
+        require(escrows[poolId] == address(0), EscrowAlreadyDeployed());
+        PoolEscrow escrow_ = new PoolEscrow{salt: bytes32(uint256(poolId))}(poolId, address(this));
+
+        escrow_.rely(root);
+        escrow_.rely(poolManager);
+        escrow_.rely(balanceSheet);
+        escrow_.rely(asyncRequests);
```

Comment:
QUESTION

Do we need to rely them? Isn't the approval enough?

Reply:
We need to rely all of these contracts because they are executing escrow calls.

Reply:
Ah, yes! the increase and so on! Perfect!